actionmailer/lib/action_mailer/parameterized.rb

```diff
@@ -161,7 +162,8 @@ def delivery_job_class
 
         def arguments_for(delivery_job, delivery_method)
           if delivery_job <= MailDeliveryJob
-            [@mailer_class.name, @action.to_s, delivery_method.to_s, params: @params, args: @args]
+            ruby2_keywords_arguments \
+              @mailer_class.name, @action.to_s, delivery_method.to_s, params: @params, args: @args
```

Comment:
This is still not future safe. Workers with old code will still break as soon we change the code to actually split the arguments. We should at least add a kwargs argument to the job with a default as `nil` right now so when we have to split the arguments for real the workers are ready to receive a pass them. But I'm not sure if we can even pass them right now, what makes this solution still not good enough.

Reply:
I'm not sure why "Workers with old code will still break as soon we change the code to actually split the arguments", but I agree that "We should at least add a kwargs argument to the job with a default" as a future plan, that is to be required if Ruby team will decide to remove internal kwargs flag (my hope that removal should not be happen since there is a serialization/deserialization problem, though).

Anyway, I'll revert this for now.

Reply:
Let's say we deploy Rails X to production, the first request running on 6.1 add a job to the queue with the arguments and keyword arguments split. The job worker that fetches that job is still using Rails X-1, given the deploy is still happening. That same worker would have code that doesn't know how to deal with the split keyword arguments so it would only pass around the regular arguments and leave the keyword arguments out of the method call.

Reply:
Thanks for the explanation. I see that concern is almost not for this change but for a future plan.

Reply:
Right. As soon we release the code to deal with this is better. It means that the entire 6.1+ releases will be able to run Ruby 3.0 without having to release a version to prepare and a version to actually add support.